### PR TITLE
Add Timestamp to Beholder attributes

### PR DESCRIPTION
**Pull request by nanchano**

### CODE REVIEW

File: pkg/beholder/message.go [hunk 1]
```diff
@@ -166,6 +169,8 @@ func OtelAttr(key string, value any) otellog.KeyValue {
 		return otellog.Bool(key, v)
 	case []byte:
 		return otellog.Bytes(key, v)
+	case time.Time:
+		return otellog.Int64(key, v.Unix())
```
Comment (cll-gg) [inline pkg/beholder/message.go:173]: In `beholder/client.go` we're setting `timestamp` to `time.Now().UTC()` which is a string representing a date time. Here however we're using the epoch timestamp. Or are these unrelated?

Comment (nanchano) [reply pkg/beholder/message.go:173]: UTC() returns a `time.Time` which can be parsed into a string or any kind of unix timestamp. I decided to use a unix timestamp here but we can render it as a string too, don't really have a preference.

File: pkg/beholder/client.go [hunk 2]
```diff
@@ -239,6 +240,11 @@ func newOtelResource(cfg Config) (resource *sdkresource.Resource, err error) {
 // Open question: what are pros/cons for using use map[]any vs use otellog.KeyValue
 func (e messageEmitter) Emit(ctx context.Context, body []byte, attrKVs ...any) error {
 	message := NewMessage(body, attrKVs...)
+
+	if _, ok := message.Attrs["timestamp"]; !ok {
+		message.Attrs["timestamp"] = time.Now().UTC()
+	}
```
Comment (jmank88) [inline pkg/beholder/client.go:246]: It looks like `log.Record`s already have a https://pkg.go.dev/go.opentelemetry.io/otel/log#Record.Timestamp and an https://pkg.go.dev/go.opentelemetry.io/otel/log#Record.ObservedTimestamp. Could we lever those built-ins instead?

Comment (pkcll) [reply pkg/beholder/client.go:246]: yep, I think I mentioned that somewhere in other PR previously

Comment (nanchano) [reply pkg/beholder/client.go]: Done

File: pkg/beholder/client.go [hunk 3]
```diff
@@ -247,10 +247,17 @@ func newOtelResource(cfg Config) (resource *sdkresource.Resource, err error) {
 // Open question: what are pros/cons for using use map[]any vs use otellog.KeyValue
 func (e messageEmitter) Emit(ctx context.Context, body []byte, attrKVs ...any) error {
 	message := NewMessage(body, attrKVs...)
+	record := message.OtelRecord()
+
+	if _, ok := message.Attrs["timestamp"]; !ok {
```
Comment (patrickhuie19) [inline pkg/beholder/client.go:252]: This feels like a separate concern than Emit - can we handle attribute hydration higher up the stack?

File: pkg/beholder/client.go (hunk 2 above)
Comment (jmank88) [reply pkg/beholder/client.go]: I expected this to be a much more simplifying change. Why do we still need to pass this around as metadata/attribute if we are using the built-in field?

**Comment by nanchano** [comment]
Closing as it won't be done

**Pull request closed without merging**